pkg/agent/test/gcs_test.go

```diff
@@ -0,0 +1,195 @@
+package test
```

Comment:
why not move your test to: `pkg/agent/watcher_test.go` as there is already a test suite that is available there, you would only need to add a mocked provider as you had done in your unit test. I think moving the providers into a package called: `mocks` might be useful however for organization. The presence of s3 mocks in `pkg/agent/watcher_test.go` seems incorrect if there are more than one protocols

. For isolation, maybe you could leave